src/bpfd.c

```diff
@@ -700,6 +700,50 @@ int main(int argc, char **argv)
 			printf("GET_KSYM_ADDR: ret=%d\n", ret);
 			if (!ret)
 				printf("%"PRIu64"\n", addr);
+		} else if (!strcmp(cmd, "GET_USYM_NAME")) {
+			int len, ret, pid, demangle;
+			uint64_t addr;
+			struct bcc_symbol sym;
+			const char *name;
+			void *usym_cache = NULL;
+
+			PARSE_FIRST_INT(pid);
+			PARSE_UINT64(addr);
+			PARSE_INT(demangle);
+
+			usym_cache = bcc_symcache_new(pid, NULL);
```

Comment:
Instead of introducing an issue (creation of cache every time), and fixing it in a later commit, you should design it correctly to begin with.